src/test/java/ltd/clearsolutions/lingualeo/client/LingualeoClientIntegrationTest.java

```diff
@@ -0,0 +1,99 @@
+package ltd.clearsolutions.lingualeo.client;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class LingualeoClientIntegrationTest {
+
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    LingualeoClient correctLingualeoClient = new LingualeoClient("vasyll.danylenko@gmail.com", "6485644Df");
+
+    LingualeoClient IncorrectLingualeoClient;
+
+    @Test
+    void auth_PutCorrectCredentials_Cookie() {
+        correctLingualeoClient = new LingualeoClient("vasyll.danylenko@gmail.com", "6485644Df");
+        correctLingualeoClient.auth();
+
+        assertNotNull(correctLingualeoClient.authCookie);
+    }
+
+    @Test
+    void auth_PutIncorrectCredentials_Null() {
+        IncorrectLingualeoClient = new LingualeoClient("vasyll.danyleil.com", "64");
+        correctLingualeoClient.auth();
+
+        assertNull(IncorrectLingualeoClient.authCookie);
```

Comment:
very strange behavior in this test
you setted new client to variable IncorrectLingualeoClient 
` IncorrectLingualeoClient = new LingualeoClient("vasyll.danyleil.com", "64");`
and then call 
`correctLingualeoClient.auth();`

and check cookies in class IncorrectLingualeoClient 
of course cookies will be null in this class(